### ISSUE TITLE
[0.11.0] Unable to form cluster on latest nightly

### ISSUE THREAD

**Issue by rossmcdonald** [influxdata/influxdb#5801]
Version information:

```
InfluxDB v0.11.0.n1456214433 (git: master 2894234b1ec225752e9ac5bcda1e5cd5a971f87f, built 2016-02-23T08:01:27.417783)
```

I'm seeing crashes when attempting to cluster three servers together (all meta/data nodes). On initial startup, all but one of the servers will crash, resulting in the first server receiving "Connection Refused" errors from the other two. When performing a `show servers` on the node that doesn't crash, it shows:

```
> show servers
name: data_nodes
----------------
id	http_addr	tcp_addr


name: meta_nodes
----------------
id	http_addr	tcp_addr
1	influx2:8091	influx2:8088
2	influx3:8091	influx3:8088
3	influx1:8091	influx1:8088
```

And the logs show:

```
[meta] 2016/02/23 16:30:31 127.0.0.1 - - [23/Feb/2016:16:30:31 +0000] POST /join HTTP/1.1 503 25 - Go 1.1 package http c1a952aa-da4a-11e5-81b9-000000000000 1.07323ms
[metastore] 2016/02/23 16:30:31 [WARN] raft: Election timeout reached, restarting election
[metastore] 2016/02/23 16:30:31 [INFO] raft: Node at influx1:8088 [Candidate] entering Candidate state
[metastore] 2016/02/23 16:30:31 [ERR] raft: Failed to make RequestVote RPC to influx2:8088: dial tcp 172.28.128.10:8088: connection refused
[metastore] 2016/02/23 16:30:31 [ERR] raft: Failed to make RequestVote RPC to influx3:8088: dial tcp 172.28.128.8:8088: connection refused
[metastore] 2016/02/23 16:30:31 [DEBUG] raft: Votes needed: 2
[metastore] 2016/02/23 16:30:31 [DEBUG] raft: Vote granted. Tally: 1
[meta] 2016/02/23 16:30:32 no leader
```

While the second and third node display the panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x58 pc=0x5b134c]

goroutine 90 [running]:
github.com/influxdata/influxdb/cluster.(*PointsWriter).MapShards(0xc20805c6e0, 0xc208222980, 0x7, 0x0, 0x0)
        /tmp/tmp.EyZeLqfQkf/src/github.com/influxdata/influxdb/cluster/points_writer.go:215 +0x6ec
github.com/influxdata/influxdb/cluster.(*PointsWriter).WritePoints(0xc20805c6e0, 0xc208222980, 0x0, 0x0)
        /tmp/tmp.EyZeLqfQkf/src/github.com/influxdata/influxdb/cluster/points_writer.go:248 +0x233
github.com/influxdata/influxdb/cmd/influxd/run.(*monitorPointsWriter).WritePoints(0xc20805c6e0, 0xc2080d7ee0, 0x9, 0x0, 0x0, 0xc208334900, 0x3, 0x3, 0x0, 0x0)
        /tmp/tmp.EyZeLqfQkf/src/github.com/influxdata/influxdb/cmd/influxd/run/server.go:755 +0xf8
github.com/influxdata/influxdb/monitor.(*Monitor).storeStatistics(0xc20808a000)
        /tmp/tmp.EyZeLqfQkf/src/github.com/influxdata/influxdb/monitor/service.go:343 +0xbf8
created by github.com/influxdata/influxdb/monitor.(*Monitor).Open
        /tmp/tmp.EyZeLqfQkf/src/github.com/influxdata/influxdb/monitor/service.go:104 +0x33c
```

All nodes have the same configuration, and have the `-join` operands set to:

```
vagrant@influx1:~$ cat /etc/default/influxdb
INFLUXD_OPTS="-join influx1:8091,influx2:8091,influx3:8091"
```

When I try starting one of the nodes that was crashed, the server that didn't crash ends up throwing the same panic, so only one node can be active at one time. For example, on initial start:

- `influx1` is up, `influx2` and `influx3` are crashed
- try bringing up `influx2` (`influx3` still down), `influx1` crashes and `influx2` stays up
- try bringing up `influx3` (`influx1` still down), `influx2` crashes and `influx3` stays up

**Comment by jwilder** [comment]
Fixed via #5800

**Comment by jwilder** [comment]
This was a regression introduced via #5757.
